From 7519af28190fc5399015116faff6e7e82825fa66 Mon Sep 17 00:00:00 2001
From: Teoh Han Hui <teohhanhui@gmail.com>
Date: Fri, 3 May 2024 03:04:25 +0800
Subject: [PATCH] Pass compulsory `mode` argument to `open` when `O_CREAT` is
 used

From `man 2 open`:

> The mode argument must be supplied if O_CREAT or O_TMPFILE is
> specified in flags; if it is not supplied, some arbitrary bytes
> from the stack will be applied as the file mode.
---
 Source/Tools/FEXLoader/FEXLoader.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/Source/Tools/FEXLoader/FEXLoader.cpp b/Source/Tools/FEXLoader/FEXLoader.cpp
index 66e38a6ca..cae5a231b 100644
--- a/Source/Tools/FEXLoader/FEXLoader.cpp
+++ b/Source/Tools/FEXLoader/FEXLoader.cpp
@@ -329,7 +329,8 @@ int main(int argc, char** argv, char** const envp) {
         LogMan::Msg::InstallHandler(FEXServerLogging::MsgHandler);
       }
     } else if (!LogFile.empty()) {
-      OutputFD = open(LogFile.c_str(), O_CREAT | O_CLOEXEC | O_WRONLY);
+      constexpr int USER_PERMS = S_IRUSR | S_IWUSR | S_IRGRP | S_IROTH;
+      OutputFD = open(LogFile.c_str(), O_CREAT | O_CLOEXEC | O_WRONLY, USER_PERMS);
     }
   }
 
-- 
2.44.0

